src/main.cpp

```diff
@@ -434,11 +437,11 @@ void run()
 
     for (int i = 0; i < timings.size(); ++i)
     {
-      // Get min/max; ignore the first result
-      auto minmax = std::minmax_element(timings[i].begin()+1, timings[i].end());
+      // Get min/max; ignore warmup iterations
+      auto minmax = std::minmax_element(timings[i].begin() + num_warmups, timings[i].end());
 
-      // Calculate average; ignore the first result
-      double average = std::accumulate(timings[i].begin()+1, timings[i].end(), 0.0) / (double)(num_times - 1);
+      // Calculate average; ignore warmup iterations
+      double average = std::accumulate(timings[i].begin() + num_warmups, timings[i].end(), 0.0) / (double)(num_times - num_warmups);
```

Comment:
For the sake of consistency across apps I think we should make the warmup iterations be additional iterations beyond the number passed to -n, so that for example -n 100 -w 10 does 10 untimed warmup iterations and then 100 timed iterations. You can implement that by just adding a `num_times += num_warmups` somewhere appropriate, maybe right after the command line input parsing.

Reply:
I included the warmup iterations into the main loop.